android/titanium/src/java/org/appcelerator/titanium/view/TiCompositeLayout.java

```diff
@@ -464,6 +472,13 @@ public WindowInsets onApplyWindowInsets(WindowInsets insets)
 			return null;
 		}
 
+		// Do not propagate given insets to child views unless the insets have changed.
+		// This greatly improves performance if we have deeply nested views since each view will trigger a relayout.
+		if (insets.equals(this.previousInsets)) {
+			return insets;
+		}
+		this.previousInsets = insets;
```

Comment:
Does this also need an Android api level guard? Would accessing it on api level < 20 cause a crash or something?

Reply:
We don't need a guard inside the `onApplyWindowInsets()` method since it has argument type `WindowInsets` which is API Level 20 only. Attempting to call `onApplyWindowInsets()` on a lower API level would crash, meaning that the API Level guard needs to be done on the caller's side.

But good question. Thanks for asking.

Reply:
That said, I'll do a quick experiment with Google's `@RequireApi(ApiLevel)` Java annotation. Android Studio's built-in linting tool is supposed to trigger compiler errors/warnings when making unguarded calls to methods like this. When we switch over to gradle, this may help reduce crashes on older OS versions.

Reply:
I wrote up a ticket [TIMOB-27437](https://jira.appcelerator.org/browse/TIMOB-27437) to implement the `@RequireApi()` annotations in the future. We can't leverage it until we support the gradle linting tool and add the Google Support Annotation library.